src/errors/NoAvailableTokenError.ts

```diff
@@ -0,0 +1,13 @@
+import { BaseIsomerError } from "@errors/BaseError"
+
+export default class NoAvailableTokenError extends BaseIsomerError {
+  timestamp: Date
+
+  constructor(
+    message = "Unable to select token for request",
+    timestamp = new Date()
```

Comment:
Sanity check here: we already have a time stamp in our logger methods, do you still want this explictedly here?

Reply:
I think it make sense to still have a timestamp for when TokenService fails to provide token, because not all errors will be logged. It can be used to get downtime in application without having to go through logs.

Reply:
> because not all errors will be logged.

wait I am confused here, if the errors are not logged, how will you know that the error occurred in the first place ya?

Reply:
Just for clarification, all possible errors that `TokenService` can encounter has already been logged on creation. For the `NoAvailableTokenError` specifically, since it is tied to the token limits of the service, I think it will be beneficial to have knowledge of which requests has failed and its timestamp in application. The logs will have another matching timestamp which provides the means of monitoring externally.